packages/@lwc/template-compiler/src/shared/renderer-hooks.ts

```diff
@@ -55,15 +61,6 @@ function checkElement(element: BaseElement, state: State): boolean {
             return state.crDirectives.has(ElementDirectiveName[dir.name]);
         });
         if (directiveMatched) {
-            // Directives that require custom renderer are not allowed on custom elements
-            // Custom element cannot be allowed to have a custom renderer hook
-            // The renderer is cascaded down from the owner(custom element) to all its child nodes who
-            // do not have a renderer specified.
-            invariant(
-                element.type !== 'Component',
-                TemplateErrors.DIRECTIVE_DISALLOWED_ON_CUSTOM_ELEMENT,
```

Comment:
Please clean up this diagnostic in template-transform.ts as well. It is not used anymore.